Delcam.ProductInterface.PowerMILL/Classes/PMFeatureGroupsCollection.cs

```diff
@@ -0,0 +1,81 @@
+﻿// **********************************************************************
+// *         © COPYRIGHT 2018 Autodesk, Inc.All Rights Reserved         *
+// *                                                                    *
+// *  Use of this software is subject to the terms of the Autodesk      *
+// *  license agreement provided at the time of installation            *
+// *  or download, or which otherwise accompanies this software         *
+// *  in either electronic or hard copy form.                           *
+// **********************************************************************
+
+using System;
+using System.Collections.Generic;
+
+namespace Autodesk.ProductInterface.PowerMILL
+{
+    /// <summary>
+    /// Represents the collection of FeatureGroup objects in the Active PowerMILL Project.
+    /// </summary>
+    public class PMFeatureGroupsCollection : PMEntitiesCollection<PMFeatureGroup>
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Initialises the item.
+        /// </summary>
+        /// <param name="powerMILL">The base instance to interact with PowerMILL.</param>
+        internal PMFeatureGroupsCollection(PMAutomation powerMILL) : base(powerMILL)
+        {
+            Initialise();
+        }
+
+        #endregion
+
+        #region Operations
+
+        /// <summary>
+        /// Initialises the collection with the items in PowerMILL.
+        /// </summary>
+        internal void Initialise()
+        {
+            if (_powerMILL.Version.Major >= 2018)
+            {
+                foreach (string name in ReadFeatureGroups())
```

Comment:
This is not going to execute in every foreach loop?

Reply:
No, that's not how they work.  The iteration function is only called once.
https://stackoverflow.com/questions/2447559/c-sharp-does-function-get-called-for-each-iteration-of-a-foreach-loop
I tested it to be doubly sure!